src/email/email.ts

```diff
@@ -0,0 +1,190 @@
+    try {
+      // Check if the path is a URL
+      if (this.isUrl(filePath)) {
+        try {
+          const response = await axios.get(filePath, { responseType: 'arraybuffer', timeout: 30000 });
+          const content = response.data;
+          
+          return Buffer.from(content).toString('base64');
+        } catch (error: unknown) {
+          if (error instanceof Error) {
+            throw new SendLayerValidationError(`Error fetching remote file: ${error.message}`);
+          }
+          throw new SendLayerValidationError('Unknown error occurred while fetching remote file');
+        }
+      }
+
+      // Try different local paths
+      const pathsToTry = [
+        filePath, // Original path
+        path.resolve(filePath), // Absolute path
+        path.join(process.cwd(), filePath), // Relative to current working directory
+        path.join(__dirname, filePath), // Relative to script directory
+        path.join(process.env.HOME || '', filePath) // Relative to user's home directory
+      ];
+
+      for (const currentPath of pathsToTry) {
+        try {
+          if (fs.existsSync(currentPath)) {
+            if (!fs.statSync(currentPath).isFile()) {
+              throw new SendLayerValidationError(`Path is not a file: ${currentPath}`);
+            }
+            try {
+              fs.accessSync(currentPath, fs.constants.R_OK);
+            } catch {
+              throw new SendLayerValidationError(`File is not readable: ${currentPath}`);
+            }
+            const content = fs.readFileSync(currentPath);
+            
+            return Buffer.from(content).toString('base64');
+          }
+        } catch (error: unknown) {
+          if (error instanceof SendLayerValidationError) {
+            throw error;
+          }
+        }
+      }
+
+      throw new SendLayerValidationError(`Attachment file not found: ${filePath}`);
+    } catch (error: unknown) {
+      if (error instanceof SendLayerValidationError) {
+        throw error;
+      }
+      if (error instanceof Error) {
+        throw new SendLayerValidationError(`Error reading attachment: ${error.message}`);
+      }
+      throw new SendLayerValidationError('Unknown error occurred while reading attachment');
+    }
+  }
+
+  async send(options: EmailOptions): Promise<SendEmailResponse> {
+    // Validate sender first
+    if (!this.validateEmail(options.from_email)) {
+      throw new SendLayerValidationError(`Invalid sender email address: ${options.from_email}`);
+    }
+
+    // Validate and process recipients
+    const toList = Array.isArray(options.to) 
+      ? options.to.map(r => this.validateRecipient(r, "recipient"))
+      : [this.validateRecipient(options.to, "recipient")];
+
+    const payload: any = {
+      From: {
+        email: options.from_email,
+        name: options?.from_name
+      },
+      To: toList,
+      Subject: options.subject,
+      ContentType: options.html ? "HTML" : "Text",
```

Comment:
Maybe it's worth hiding these "magic" strings behind enums? That would make it more accessible to IDEs.